src/v0/destinations/salesforce/transform.js

```diff
@@ -106,24 +115,30 @@ async function getSaleforceIdForRecord(
   identifierType,
   identifierValue,
   destination,
+  authorizationFlow,
 ) {
   const objSearchUrl = `${authorizationData.instanceUrl}/services/data/v${SF_API_VERSION}/parameterizedSearch/?q=${identifierValue}&sobject=${objectType}&in=${identifierType}&${objectType}.fields=id,${identifierType}`;
+  const finalHeader =
+    authorizationFlow === 'oauth'
+      ? { Authorization: `Bearer ${authorizationData.token}` }
+      : { Authorization: authorizationData.token };
   const { processedResponse: processedsfSearchResponse } = await handleHttpRequest(
     'get',
     objSearchUrl,
     {
-      headers: { Authorization: authorizationData.token },
+      headers: finalHeader,
     },
     {
       destType: 'salesforce',
       feature: 'transformation',
     },
   );
-  if (processedsfSearchResponse.status !== 200) {
+  if (!isHttpStatusSuccess(processedsfSearchResponse.status)) {
```

Comment:
@yashasvibajpai Can you guys confirm if this change would not create any problem ?

Reply:
Yes, seems fine
200, 201 and 204 will be handled.